### PR TITLE
Added embedding support.

### PR DESCRIPTION
**Pull request by Alasnkz**
This pull request adds embedding support.

Example code:
```cpp
public DCC_OnMessageCreate(DCC_Message:message)
{
    new content[DCC_ID_SIZE], DCC_Channel:channel;
    DCC_GetMessageContent(message, content);
    DCC_GetMessageChannel(message, channel);
   
	if(!strcmp(content[0], "msg"))
	{
                new DCC_Embed:embed = DCC_CreateEmbedMessage("hello, world!", "this is a test", "https://google.com", .thumbnail_url="https://avatars3.githubusercontent.com/u/1799378?s=460&u=e6d0167651af9fd760c24e63661b27d4ef771fdc&v=4");
		for(new i = 0; i < 12; i++)
		{
	 		new str[50];
	 		format(str, sizeof str, "field %d", i);
	   		DCC_AddEmbedField(embed, str, "test", true);
		}
		DCC_SendEmbedMessage(channel, embed, "test");
	}
    return 1;
}
```

Result:

### CODE REVIEW

**Comment by spacemud** [comment]
:shipit:

File: discord-connector.inc.in [hunk 1]
```diff
@@ -179,7 +179,18 @@ stock DCC_ClearBotActivity()
...
+// embedded messages
+native DCC_Embed:DCC_CreateEmbedMessage(const title[] = "", const description[] = "", const url[] = "", const timestamp[] = "", color = 0, const footer_text[] = "", const footer_icon_url[] = "", const footer_proxy_icon_url[] = "", const thumbnail_url[] = "", const thumbnail_proxy_url[] = "", thumbnail_height = 0, thumbnail_width = 0);
+native DCC_DeleteEmbedMessage(DCC_Embed:embed);
+native DCC_SendEmbedMessage(DCC_Channel:channel, DCC_Embed:embed, const message[] = "", const callback[] = "", const format[] = "", {Float, _}:...);
+native DCC_AddEmbedField(DCC_Embed:embed, const name[], const value[], bool:inline = false);
+native DCC_SetEmbedTitle(DCC_Embed:embed, const title[]);
+native DCC_SetEmbedDescription(DCC_Embed:embed, const description[] = "");
+native DCC_SetEmbedUrl(DCC_Embed:embed, const url[] = "");
+native DCC_SetEmbedTimestamp(DCC_Embed:embed, const timestamp[] = "");
+native DCC_SetEmbedColor(DCC_Embed:embed, color = 0);
```
Comment (maddinat0r) [inline discord-connector.inc.in:191]:
```
native DCC_SetEmbedColour(DCC_Embed:embed, colour) = DCC_SetEmbedColor; // oi m8
```

**Comment by Alasnkz** [comment]
I must have automatically assumed that the "image" embed option was for actual file image uploads and confused myself. I'll work on this now. Thanks for the review.

**Comment by maddinat0r** [comment]
Merged. Thanks!

**Pull request merged**